### PR TITLE
update alertmanager dockerfile

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Details

Modifies entire alertmanager Dockerfile to support ca certificates chain update based on enable and disable flag 

## Related Issues

https://github.com/astronomer/issues/issues/6334

## Checklist

- [ ] version.txt was updated

### CODE REVIEW

File: alertmanager/Dockerfile [hunk 1]
```diff
@@ -14,10 +14,33 @@
 # limitations under the License.
 
 # https://hub.docker.com/r/prom/alertmanager/tags?page=1&ordering=last_updated
-FROM prom/alertmanager:v0.27.0
-LABEL maintainer="Astronomer <humans@astronomer.io>"
+FROM prom/alertmanager:v0.27.0 AS UPSTREAM
 
+FROM quay.io/astronomer/ap-base:3.18.6-2
```
Comment (danielhoherd) [inline alertmanager/Dockerfile:19]: @pgvishnuram Why did we change this to a multi-stage build?

Comment (pgvishnuram) [reply alertmanager/Dockerfile:19]: im unable to find a way to add update-ca-certificates in busybox base image

Comment (danielhoherd) [reply alertmanager/Dockerfile:19]: That makes sense. This looks like the Dockerfile: https://github.com/prometheus/alertmanager/blob/main/Dockerfile I think using a two-stage build is fine in this case.

File: alertmanager/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,21 @@
+#!/bin/sh
+
+set -e
+
+# In production, we mount the registry TLS certificate
+# into /usr/local/share/ca-certificates. This completes
+# the certificate installation so we can trust the registry.
+if [ "$UPDATE_CA_CERTS" = "true" ]; then
+echo "Running update-ca-certificates"
+update-ca-certificates
```
Comment (danielhoherd) [inline alertmanager/entrypoint.sh:10]:
```suggestion
  echo "Running update-ca-certificates"
  update-ca-certificates
```

File: alertmanager/Dockerfile [hunk 3]
```diff
@@ -14,10 +14,33 @@
...
+COPY --from=UPSTREAM /bin/alertmanager /usr/local/bin
+COPY --from=UPSTREAM /etc/alertmanager/alertmanager.yml   /etc/alertmanager/alertmanager.yml
+COPY --chown=65534:65534 entrypoint.sh /entrypoint.sh
+
+# Adopt openshift permissions
+RUN chown -R 65534:65534 /etc/ssl/ \
+    && chown -R 65534:0 /etc/ssl/certs \
+    && chmod -R ug+rwx /etc/ssl/certs
+
+
```
Comment (danielhoherd) [inline alertmanager/Dockerfile:38]:
```suggestion
# Copied from https://github.com/prometheus/alertmanager/blob/main/Dockerfile
```

**Review by danielhoherd — APPROVED** [review]
Seems fine to me. Having tests would be great, but I don't think that should block merging this.

**Pull request merged**